dotnet/src/Extensions/PromptTemplates.Handlebars/Helpers/KernelHelpers/KernelFunctionHelpers.cs

```diff
@@ -198,9 +200,29 @@ private static void ProcessPositionalArguments(KernelFunctionMetadata functionMe
     private static object? ParseResult(FunctionResult result)
     {
         var resultAsObject = result.GetValue<object?>();
-        if (result.ValueType is not null && resultAsObject is not null && result.ValueType == typeof(OpenAIChatMessageContent))
+        if (result.ValueType is not null && resultAsObject is not null)
         {
-            resultAsObject = ((OpenAIChatMessageContent)resultAsObject).Content;
+#pragma warning disable SKEXP0042 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+            if (resultAsObject is RestApiOperationResponse restApiOperationResponse)
+            {
+                if (restApiOperationResponse.ExpectedSchema is not null)
```

Comment:
Is this check used to determine if the response is expected to be JSON? Would it make sense to test against the `restApiOperationResponse.ContentType` instead?

Either way, would you mind adding a few comments in around the various code branches in this method to explain what the check is telling us.

Reply:
I changed the check to `if (string.Equals(restApiOperationResponse.ContentType, "application/json", StringComparison.OrdinalIgnoreCase))`

And added more comments!